h2o-algos/src/main/java/hex/glm/GLM.java

```diff
@@ -1572,16 +1659,51 @@ public final class GLMDriver extends Driver implements ProgressMonitor {
     private transient Cholesky _chol;
     private transient L1Solver _lslvr;
 
+    private double[] constraintGLM_solve(GramGrad  gram) {
+      if (!_parms._intercept) throw H2O.unimpl();
```

Comment:
My basic assumption is that 

$$\hat{y}=\mathbb{X}\beta$$ 

and $\mathbb{X}$ is a matrix with one column full of ones and that's used to estimate the intercept in $\beta$. Naively, I don't see a reason why there would have to be different treatment of this $\beta$ with respect to some constraints. If we get a collinear column we would remove it and if we don't want to estimate intercept we can set the "intercept" column in $\mathbb{X}$ to zeros which would be by definition a colinear column (any column minus itself would create this column) so the reasonable action seems to me to treat it as colinear column which I think is already implemented.

Is it just too much work for little gain or is there some underlying problem?

I'm just asking since I want to understand the things I review.

Reply:
Your formula was implemented with the concept of an intercept term.  The sizing of the gram matrix and the generation of xy terms.  

In general, we can implement GLM without the intercept term as well.  However, with the solution with the intercept is more general because allowing the intercept term and if it come out to be zero, then it is equivalent to not having intercept.

Reply:
I wonder if we should advise users to not use `intercept=false` when using constrained GLM (e.g. emitting error/warning in `init`) or if we can map `intercept=false` to a constraint `intercept==0`. Either options is ok for me. The former is good in a sense that it fails fast and informs AutoML or Grid Seach that the parameter configuration was incorrect. The latter seems more hacky but I think it would still be a possible solution.

Reply:
I move the check to init before expensive